src/services/completions.ts

```diff
@@ -764,7 +764,7 @@ namespace ts.Completions {
         if (preferences.includeCompletionsWithClassMemberSnippets &&
             preferences.includeCompletionsWithInsertText &&
             completionKind === CompletionKind.MemberLike &&
-            isClassLikeMemberCompletion(symbol, location)) {
+            isClassLikeMemberCompletion(symbol, location) && !hasInitializer(location.parent)) {

```

Comment:
Seems like you want something more like

```ts
location.parent.children().some(child => child.pos > location.end)
```

since you could lose a more-specific type annotation, or trivia. For example:

```ts
////interface I {
////    prop: string;
////}
////class C implements I {
////    public pr/**/: "foo" | "bar";
////}
```